File: delete_account.py
import mysql.connector

def delete_account(username, password):
    try:
        
        db_connection = mysql.connector.connect(
            host="localhost",
            user="rpatel245",
            password="rpatel245",
            database="rpatel245"
        )
        
        cursor = db_connection.cursor()
        
        # verify user creds
        query = "SELECT * FROM users WHERE username = %s AND password = %s"
        cursor.execute(query, (username, password))
        user = cursor.fetchone()
        
        if user:
            # account deletion confirmation
            confirmation = input("Are you sure you want to delete your account? (yes/no): ")
            if confirmation.lower() == "yes":
                # Delete user account
                delete_query = "DELETE FROM users WHERE username = %s"
                cursor.execute(delete_query, (username,))
                db_connection.commit()
                print("Account deleted successfully")
            else:
                print("Account deletion canceled")
        else:
            print("Invalid username or password")
            
    except mysql.connector.Error as error:
        print(f"Error: {error}")

# user and pass prompt
username = input("Enter your username: ")
password = input("Enter your password: ")


delete_account(username, password)

File: delete_account.php
<?php
session_start();

// Check if the user is logged in.
if (!isset($_SESSION['username'])) {
    echo "You need to log in to delete your account.";
    exit();
}

$host = "localhost";
$user = "rpatel245";
$pass = "rpatel245";
$dbname = "rpatel245";

$conn = new mysqli($host, $user, $pass, $dbname);

if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}

$username = $_SESSION['username'];

// First, delete dependent records in the saved_recipes table
$stmt = $conn->prepare("DELETE FROM saved_recipes WHERE user_id IN (SELECT id FROM users WHERE username=?)");
$stmt->bind_param("s", $username);
if (!$stmt->execute()) {
    echo "Error clearing user related data: " . $stmt->error;
    $stmt->close();
    $conn->close();
    exit();
}
$stmt->close();

// Now, delete the user
$stmt = $conn->prepare("DELETE FROM users WHERE username=?");
$stmt->bind_param("s", $username);

if ($stmt->execute() === TRUE) {
    echo "Account deleted successfully.";

    session_unset();
    session_destroy();
    header("Location: login.php");
    exit();
} else {
    echo "Error deleting account: " . $conn->error;
}

$stmt->close();
$conn->close();
?>
